frontend/src/router/index.ts

```diff
@@ -69,11 +75,19 @@ const routes = [
         component: ManagePermissionsView,
         beforeEnter: async (to: any) => {
             // Requires fetching applications the user administers.
-            await fetchApplications();
-            const userRoleAssignments = await fetchUserRoleAssignments(
-                selectedApplication.value?.application_id
-            );
-            Object.assign(to.meta, { userRoleAssignments: userRoleAssignments });
+            try {
+                await fetchApplications();
+                // if user already selected an application in cache
+                const userRoleAssignments = await fetchUserRoleAssignments(
+                    selectedApplication.value?.application_id
+                );
+                Object.assign(to.meta, {
+                    userRoleAssignments: userRoleAssignments,
+                });
+            } catch (error) {
+                setToastErrorMsg(error);
```

Comment:
we have to catch the error on api calls in the beforeEnter method, otherwise it will throw the router error, and it can't be handled. And we also can't popup the toast at this step because it's not in any of the router yet, we can only store the error messages and popup whenever enter the router page.

Reply:
one design question, so when the user login but has no role, do we show him empty table on the manage permission page, or we direct him back to landing page?